rotkehlchen/history/events.py

```diff
@@ -536,3 +538,24 @@ def fail_history_cb(error_msg: str) -> None:
             defi_events,
             ledger_actions,
         )
+
+    def query_kraken_staking_events(
```

Comment:
I was not sure where to put the function that executes this logic and I went with this one. Also this functions queries the kraken ledger in the full range when the cache is invalidated. I'm not sure this is something you agree with. This makes sense with what we do with trades or deposits. Also if it was recently queried it won't have a big inpact

Reply:
This should not be here. You are mixing things here and breaking down well established barriers between modules.

This should be a method of the Kraken object. The exchanges list logic should be in the api/rest.py method.